Website files v2

staceynduta opened the pull request:
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

staceynduta committed:
Add files via upload

19DennisBrown committed:
Added client sub-folder that contains React inside client folder.

PeterMaks committed:
Added folder and subfolder for MC Studio Version 2.1 website assets and documentation and Subfolders for the assets and documentation of each webpage.

19DennisBrown committed:
Removed client react folder, harmonised new changes in folders.

19DennisBrown committed:
Added client ReactJS

19DennisBrown committed:
Additions

-Added image to HomePage and Profile page
-Implemented couresel on DirectoryPage

19DennisBrown committed:
Icons and Fonts

Added social icons to footer and changed the font to Roboto

PeterMaks committed:
Update to projects page and  Case Studies

PeterMaks committed:
Merge branch 'Developersden25' of https://github.com/Kijani-Support/mc-studio into Developersden25

19DennisBrown committed:
Readded Profile Page

19DennisBrown committed:
Merge branch 'Developersden25' of https://github.com/Kijani-Support/mc-studio into Developersden25

PeterMaks committed:
Add ArticleDetail page and update routing in App component; modify NavBar links

PeterMaks committed:
Cahnged backround to Globe projection, Working on the Globe shader

PeterMaks committed:
Added G&W globe image

PeterMaks committed:
Add Services page and update App routing; enhance Projects page styling and functionality

PeterMaks committed:
Refactor code structure for improved readability and maintainability
improoved on Navbar design

PeterMaks committed:
Refactor NavBar component to accept props for dark mode and theme toggle; update ArticleDetail page to use NavBar

PeterMaks committed:
update toggle light/dark mode in article detail

PeterMaks committed:
Update  Service page nd case studies page to have Dark modes

PeterMaks committed:
Updated SearCh and Query Functions

PeterMaks committed:
Added a Global Theme Context to check if the pages are in Light and Dark Mode
Impemented Search and Sort Feature in Case studies
fixed Icons In the Startups Page

TO DO:
- Fix the footer on the Projects Page
-Fix Light and Dark Mode In the Home And directory pages
-Fix the Intereactive Globe elements

PeterMaks committed:
Optimised dark mode for multiple pages

PeterMaks committed:
Optimised dark mode to Projeccts page

PeterMaks committed:
Updated navbar to show current user page

PeterMaks committed:
Removed footer and changed 3d globe with 2d map
Need To FIX:
-Points dissapearing issue
-Text rendering on zoom
-Plus mutiple other faults

PeterMaks commented on Version2.1 mcstudio website/MC WEBSITE Assets and Documentation/client/src/pages/Services.jsx:
```diff
@@ -0,0 +1,333 @@
+  Facebook, Twitter, Instagram, Linkedin, Youtube, 
+  Lightbulb, Palette, Code, Users, CheckCircle2, ChevronDown, ChevronUp 
+} from 'lucide-react';
+import { useTheme } from '../components/context/ThemeContext'; // <-- Import the hook
```
'Use /Context' instead of '/context' due to some issues in naming the File

PeterMaks committed:
updated some iport statements

PeterMaks committed:
Updated  Projects page and Context imports for all pages

PeterMaks committed:
Added server folder with README.md file to the MC WEBSITE Assets and Documentation folder.

PeterMaks committed:
Added working COBE 3d Globe  Library for interactive projects map

PeterMaks committed:
Improoved dark mode for directory page and optimised 3d Globe

PeterMaks committed:
improoved functionality

PeterMaks committed:
Improoved Globe Lerp animation

PeterMaks committed:
Fix: Labelling on Globe

PeterMaks committed:
package update

PeterMaks commented on Version2.1 mcstudio website/MC WEBSITE Assets and Documentation/client/src/components/NavBar.jsx:
```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { Link, useLocation } from 'react-router-dom'; // <-- Import useLocation
+import { Sun, Moon } from 'lucide-react';
+import { useTheme } from '../components/context/ThemeContext';
```
Vite is case sensitive when updating the Npm packages resorts to naming the "Context" folder with a capital "C"(folder was renamed)

PeterMaks committed:
Import statement issue swith vite

PeterMaks committed:
Using Absolute directory imports for the dark theme for Navbar and Footers

PeterMaks committed:
import statements refactoring

PeterMaks committed:
globe to homepage

PeterMaks committed:
Adjust globe scale

PeterMaks committed:
Adjust scales and dpi of globes

PeterMaks committed:
import statement edit

PeterMaks committed:
Added file to gitignore

PeterMaks committed:
mobile browser intergration

PeterMaks committed:
Adding netlify configs for testing

PeterMaks committed:
refine import statements

PeterMaks committed:
Update

PeterMaks committed:
Rename folder to match

PeterMaks committed:
Add ScrollToTop component and integrate it into App component

PeterMaks committed:
Merge branch 'Developersden25' of https://github.com/Kijani-Support/mc-studio into Developersden25

PeterMaks committed:
Adding links to footers

PeterMaks committed:
Beckend Brevo intergration (WIP) Routing changes in main pages

PeterMaks committed:
Accidentally stripped tailwind css from everything

PeterMaks committed:
Working Brevo intergration server side

PeterMaks committed:
Added brevo intergration on required pages

PeterMaks committed:
.env template contact me for API key

PeterMaks committed:
warning

PeterMaks committed:
edit

PeterMaks committed:
Removed packages i.e. React:
-Simple maps
-Globe Gl
-Lucide react

And optimised sections of the code and the packages used

PeterMaks committed:
typo fixing

PeterMaks committed:
Dependency on Lucide react solved

19DennisBrown committed:
Added links to case studies.

19DennisBrown committed:
Merge branch 'Developersden25' of https://github.com/Kijani-Support/mc-studio into Developersden25

PeterMaks committed:
Edit netlify toml

PeterMaks committed:
minor change

PeterMaks committed:
configs for test server

PeterMaks committed:
error fix

PeterMaks committed:
update toml

PeterMaks committed:
idk what im doing

PeterMaks committed:
input

PeterMaks committed:
ver test a

PeterMaks committed:
test b

PeterMaks committed:
test c

PeterMaks committed:
refine routes to enable use of express

PeterMaks committed:
test d

PeterMaks committed:
Test E

PeterMaks committed:
Changed Base directory for netlify easy deployment

PeterMaks committed:
update Toml

PeterMaks committed:
Test ver 1.2

PeterMaks committed:
containing functions within the netlify api.js to allow esbuild package not to ruin everything

PeterMaks committed:
mout routers update

PeterMaks committed:
Improove images and assets on certain pages
